mockgen/package_mode.go

```diff
@@ -147,7 +147,7 @@ func (p *packageModeParser) loadPackage(packageName string) (*packages.Package,
 	}
 
 	cfg := &packages.Config{
-		Mode:       packages.NeedDeps | packages.NeedImports | packages.NeedTypes | packages.NeedTypesInfo | packages.NeedEmbedFiles | packages.LoadSyntax,
+		Mode:       packages.LoadSyntax,
```

Comment:
This was simplified according to https://pkg.go.dev/golang.org/x/tools/go/packages#pkg-constants, which states that `LoadSyntax` is essentially all other options except `NeedDeps` and `NeedEmbedFiles`.

Not sure why `NeedEmbedFiles` was used here, but `NeedDeps` is the biggest performance hit and the tests are still passing with it being removed.

Reply:
```
2025/03/18 13:14:21 internal error: package "encoding/json" without types was imported from "<package>"
```

We've just run into the (in our case) 2x performance degradation (15m -> 30m), unfortunately this change (specifically the removal of `NeedDeps`) doesn't seem to work as we're unable to find types for the third party packages that we're using.